Extension/src/packageVersion.ts

```diff
@@ -50,8 +50,8 @@ export class PackageVersion {
             this.suffix === other.suffix && this.suffixVersion === other.suffixVersion;
     }
 
-    public isGreaterThan(other: PackageVersion, suffixStr: string = 'insiders'): boolean {
-        if ((this.suffix && !this.suffix.startsWith(suffixStr)) || (other.suffix && !other.suffix.startsWith(suffixStr))) {
+    public isGreaterThan(other: PackageVersion, suffixStr: string | undefined): boolean {
```

Comment:
Can you explain what is wrong with this function that requires a re-write?  I would rather we not change this functionality.

Reply:
I changed this for when I didn't want to consider the suffix when comparing the VsCode versions, 
because adding the input "insider" here didn't make sense to me, So I changed the first line.
However, I realized I don't need to change the rest of the function. 
For example, I want this:
"1.44.0.master" equal to "1.44.0"
"1.44.0.insdier" equal to "1.44.0"
"1.44.0"            equal to "1.44.0"

Reply:
There are a few things to think about to achieve this behavior.
1. `undefined` and `""` are both "falsy" values, and the comparision you've used would allow you to get the same behavior without adding `| undefined` to the type definition in the signature.
2. Changing the behavior of a public function based on magic values like `undefined` and `""` can result in confusion in the future and should be avoided when possible. Consider adding a new versions of the function that encode what you want in the function name and then have the forward to a private function that understands the magic values. For example:
```ts
public isGreaterThan(other: PackageVersion, suffixStr: string): boolean {
    return isGreaterThanInternal(other, suffixStr);
}

public isMajorMinorPatchGreaterThan(other: PackageVersion): boolean {
    return isGreaterThanInternal(other, "");
}

private isGreaterThanInternal(other: PackageVersion, suffixStr: string): boolean {
    // The real work happens here and suffixStr === "" is documented as a special thing that changes the behavior.
}
```
If you don't do it that way, at the very least, suffixStr === "" needs to be clearly documented as a special value that changes the behavior of the function.

Reply:
Thank you for the detailed explanation. 
I created different functions for different behaviors of `isGreaterThan` for vscode package and extension package.